File: matlab/Chp08/euler.m
% Numerical Mathematics and Computing, Fourth Edition
% Ward Cheney & David Kincaid
% Brooks/Cole Publ. Co.
% (c) 1999
% ISBN/ISSN:  0-534-35184-0
%
% file:  euler.m
%
% Euler's method:
% The function f is defined in the M-file 'f.m'.

a = 0, b = 4, n = 20,
h = (b - a)/n
t = a
T = t
x = 1.0
X = x
for  i=1:n
 x = x + h*f(t,x)
 t = t + h
 X = [X, x]
 T = [T, t]
end
S = a:h/4:b
Y = exp(S)
plot(T,X,'o', S,Y,'-')

File: f90code/Chp08/euler.f90
!
! Numerical Mathematics and Computing, Third Edition
! Ward Cheney & David Kincaid
! Brooks/Cole Publ. Co.
! ISBN 0-534-20112-1
! Copyright (c) 1994.  All rights reserved.
! For educational use with the Cheney-Kincaid textbook.
! Absolutely no warranty implied or expressed.
!
! Section 8.1
!
! File: euler.f90
!
! Euler's method for solving an ordinary differential equation

program main
      real :: a= 1.0, b= 2.0, x=-4.0, h, t, f
      integer :: n= 100, k

      h = (b - a)/ real(n)
      t = a 
      print *,t,x 
      do k = 1,n
         x = x + h*f(t,x)
         t = t + h 
         print*, "this is k,t,x", k,t,x 
      end do
end program main

      function f(t,x)
      real, intent(in)::x, t
      f = 1.0 + x*x + t**3    
      end function f

                                                                      




